packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -154,6 +154,11 @@ export function inferColumns(rows: GridRowsProp): SerializableGridColumns {
   return Object.entries(firstRow).map(([field, value]) => {
     return {
       field,
+      headerName:
+        // turn field into a human readable name by capitalizing the first letter and replacing camel case with spaces
+        field[0].toUpperCase() + field.slice(1).replace(/([A-Z])/g, ' $1'),
```

Comment:
can we rely on this for every type of data? what if an API returns column names that aren't camel case?

Reply:
If the name isn't in camel case it would return the name with the first letter capitalized

Reply:
yeah you're right. sounds good!